include/nfc/t4t/hl_procedure.h

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef NFC_T4T_HL_PROCEDURE_
+#define NFC_T4T_HL_PROCEDURE_
+
+/**
+ * @file
+ * @defgroup nfc_t4t_hl_procedure NFC T4T High Level Procedure
+ * @{
+ * @brief NFC T4T High Level Procedure API
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/types.h>
+#include <nfc/t4t/cc_file.h>
+
+/**@brief Type of selection procedure. */
+enum nfc_t4t_hl_procedure_select {
+	/** NDEF App select. */
+	NFC_T4T_HL_PROCEDURE_NDEF_APP_SELECT,
+
+	/** CC file select. */
+	NFC_T4T_HL_PROCEDURE_CC_SELECT,
+
+	/** NDEF file select. */
+	NFC_T4T_HL_PROCEDURE_NDEF_FILE_SELECT
+};
+
+/**@brief NFC T4T High Level Procedure callback structure.
+ *
+ * This structure is used to control command exchenge with
+ * NFC Type 4 Tag.
+ */
+struct nfc_t4t_hl_procedure_cb {
+	/**@brief HL Procedure selected callback.
+	 *
+	 * The selection of applications or files is completed.
+	 *
+	 * @param[in] type Type of selection command.
+	 */
+	void (*selected)(enum nfc_t4t_hl_procedure_select type);
+
+	/**@brief HL Procedure CC file read callback.
+	 *
+	 * The CC file of Type 4 Tag read operation is
+	 * completed successfully.
+	 *
+	 * @param[in] cc Pointer to CC file data.
+	 */
+	void (*cc_read)(struct nfc_t4t_cc_file *cc);
+
+	/**@brief HL Procedure NDEF file read callback.
+	 *
+	 * The NDEF file of Type 4 Tag read operation is
+	 * completed successfully.
+	 *
+	 * @param[in] file_id File Identifier
+	 * @param[in] data Pointer to received NDEF file data
```

Comment:
I added some more documentation to `nfc_t4t_hl_procedure_ndef_read` and `nfc_t4t_hl_procedure_ndef_update` according to buffer usage but them. In case on error on lower layer you got a callback from it or `nfc_t4t_hl_procedure_on_data_received` will return with error